pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -151,10 +153,41 @@ func (a *Azure) ValidateStore(store esv1beta1.GenericStore) error {
 	return nil
 }
 
-// Empty GetAllSecrets.
+// Implements store.Client.GetAllSecrets Interface.
+// Retrieves a map[string][]byte with the secret names as key and the secret itself as the calue.
 func (a *Azure) GetAllSecrets(ctx context.Context, ref esv1beta1.ExternalSecretFind) (map[string][]byte, error) {
-	// TO be implemented
-	return nil, fmt.Errorf("GetAllSecrets not implemented")
+	basicClient := a.baseClient
+	secretsMap := make(map[string][]byte)
+	checkTags := len(ref.Tags) > 0
+	checkName := ref.Name != nil && len(ref.Name.RegExp) > 0
+
+	secretListIter, err := basicClient.GetSecretsComplete(context.Background(), *a.provider.VaultURL, nil)
+
+	if err != nil {
+		return nil, err
+	}
+	for secretListIter.NotDone() {
+		secretList := secretListIter.Response().Value
+		for _, secret := range *secretList {
+			ok, secretName := isValidSecret(checkTags, checkName, ref, secret)
+			if !ok {
+				continue
+			}
+
+			secretResp, err := basicClient.GetSecret(context.Background(), *a.provider.VaultURL, secretName, "")
+			secretValue := *secretResp.Value
+
+			if err != nil {
+				return nil, err
+			}
```

Comment:
```suggestion
			secretResp, err := basicClient.GetSecret(context.Background(), *a.provider.VaultURL, secretName, "")
			if err != nil {
				return nil, err
			}
		        secretValue := *secretResp.Value

```
because `basicClient.GetSecret()` fails (due to lack of `get` permissions, for instance) then `secretResp.Value` will be `nil`, leading to a `panic`.

Reply:
This was the only issue I've seen with the code so far. Basically, if someone sets `list` permissions but no `get` permissions, we would have a panic.

Reply:
I fixed those. I decided not to "add" anything to the error sent by the provider. I think that, apart for the panic we were seeing, it was just adding noise to the log. What needed to be fixed was the permission setup in the provider, there's no need wrap anything around it.